hazelcast-sql/src/test/java/com/hazelcast/sql_slow/MapIndexScanPMigrationStressTest.java

```diff
@@ -99,62 +99,72 @@ public void stressTest_sorted() throws InterruptedException {
         IndexConfig indexConfig = new IndexConfig(IndexType.SORTED, "this").setName(randomName());
         map.addIndex(indexConfig);
 
-        MutatorThread mutator = new MutatorThread(1000L);
-        mutator.start();
-
-        assertRowsOrdered("SELECT * FROM " + MAP_NAME + " ORDER BY this DESC", expected);
+        MutatorThread mutator = new MutatorThread(2000L);
+        assertRowsOrdered("SELECT * FROM " + MAP_NAME + " ORDER BY this DESC", expected, mutator);
 
         mutator.terminate();
         mutator.join();
+        assertThat(mutatorException.get()).isNull();
     }
 
     private class MutatorThread extends Thread {
-
-        private volatile boolean active = true;
+        private boolean firstLaunch = true;
+        private boolean active = true;
         private final long delay;
 
         private MutatorThread(long delay) {
             this.delay = delay;
         }
 
-        private void terminate() {
+        private synchronized void terminate() {
```

Comment:
This is the only place where we synchronize. So there will be no two threads setting `active` to false concurrently. But we read it without synchronizing on the same object, the reading could be cached and we'll never observe the changed value. `volatile` was better.
Was there actually a need to change this?

Reply:
>  the reading could be cached and we'll never observe the changed value.
Oh, I forget about the caching. Totally agree, thanks.